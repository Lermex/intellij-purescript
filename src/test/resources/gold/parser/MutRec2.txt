Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSImportModuleNameImpl(ImportModuleName)
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('Prelude')
          PsiElement(whitespace)('\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PSImportDeclarationImpl(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSImportModuleNameImpl(ImportModuleName)
          PSQualifiedImpl(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
            PsiElement(proper name)('Console')
          PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSPositionedDeclarationRefImpl(PositionedDeclarationRef)
          PSValueRefImpl(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSDataDeclarationImpl(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('A')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('A')
            PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PSTypeConstructorImpl(TypeConstructor)
            PSQualifiedImpl(Qualified)
              PSProperNameImpl(ProperName)
                PsiElement(proper name)('B')
                PsiElement(whitespace)('\n\n')
      PSDataDeclarationImpl(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('B')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('B')
            PsiElement(whitespace)(' ')
        PSTypeAtomImpl(TypeAtom)
          PSTypeConstructorImpl(TypeConstructor)
            PSQualifiedImpl(Qualified)
              PSProperNameImpl(ProperName)
                PsiElement(proper name)('A')
                PsiElement(whitespace)('\n\n')
      PSNamedBinderImpl(NamedBinder)
        <empty list>
      PsiElement(foreign)('foreign')
      PsiElement(whitespace)(' ')
      PsiElement(import)('import')
      PsiElement(whitespace)(' ')
      PSExternDataDeclarationImpl(ExternDataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PsiElement(proper name)('S')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSFunKindImpl(FunKind)
          PSTypeConstructorImpl(TypeConstructor)
            PSQualifiedImpl(Qualified)
              PSProperNameImpl(ProperName)
                PsiElement(proper name)('Type')
                PsiElement(whitespace)('\n\n')
      PSTypeDeclarationImpl(TypeDeclaration)
        PSTypeAnnotationNameImpl(TypeAnnotationName)
          PsiElement(identifier)('f')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PSTypeConstructorImpl(TypeConstructor)
              PSQualifiedImpl(Qualified)
                PSProperNameImpl(ProperName)
                  PsiElement(proper name)('A')
                  PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSTypeImpl(Type)
            PSTypeAtomImpl(TypeAtom)
              PSTypeConstructorImpl(TypeConstructor)
                PSQualifiedImpl(Qualified)
                  PSProperNameImpl(ProperName)
                    PsiElement(proper name)('S')
                    PsiElement(whitespace)('\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PsiElement(identifier)('f')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSCaseImpl(Case)
              PsiElement(case)('case')
              PsiElement(whitespace)(' ')
              PSValueImpl(Value)
                PSPrefixValueImpl(PrefixValue)
                  PSVarImpl(Var)
                    PSQualifiedImpl(Qualified)
                      PsiElement(identifier)('a')
                      PsiElement(whitespace)(' ')
              PsiElement(of)('of')
              PsiElement(whitespace)(' ')
              PSCaseAlternativeImpl(CaseAlternative)
                PSValueImpl(Value)
                  PSPrefixValueImpl(PrefixValue)
                    PSConstructorImpl(Constructor)
                      PSQualifiedImpl(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('A')
                          PsiElement(whitespace)(' ')
                    PSVarImpl(Var)
                      PSQualifiedImpl(Qualified)
                        PsiElement(identifier)('b')
                        PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSValueImpl(Value)
                  PSPrefixValueImpl(PrefixValue)
                    PSVarImpl(Var)
                      PSQualifiedImpl(Qualified)
                        PsiElement(identifier)('g')
                        PsiElement(whitespace)(' ')
                    PSVarImpl(Var)
                      PSQualifiedImpl(Qualified)
                        PsiElement(identifier)('b')
                        PsiElement(whitespace)('\n\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PsiElement(identifier)('g')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('b')
        PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSCaseImpl(Case)
              PsiElement(case)('case')
              PsiElement(whitespace)(' ')
              PSValueImpl(Value)
                PSPrefixValueImpl(PrefixValue)
                  PSVarImpl(Var)
                    PSQualifiedImpl(Qualified)
                      PsiElement(identifier)('b')
                      PsiElement(whitespace)(' ')
              PsiElement(of)('of')
              PsiElement(whitespace)(' ')
              PSCaseAlternativeImpl(CaseAlternative)
                PSValueImpl(Value)
                  PSPrefixValueImpl(PrefixValue)
                    PSConstructorImpl(Constructor)
                      PSQualifiedImpl(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('B')
                          PsiElement(whitespace)(' ')
                    PSVarImpl(Var)
                      PSQualifiedImpl(Qualified)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSValueImpl(Value)
                  PSPrefixValueImpl(PrefixValue)
                    PSVarImpl(Var)
                      PSQualifiedImpl(Qualified)
                        PsiElement(identifier)('f')
                        PsiElement(whitespace)(' ')
                    PSVarImpl(Var)
                      PSQualifiedImpl(Qualified)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)('\n\n')
      PSTypeDeclarationImpl(TypeDeclaration)
        PSTypeAnnotationNameImpl(TypeAnnotationName)
          PsiElement(identifier)('showN')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSTypeImpl(Type)
          PSTypeAtomImpl(TypeAtom)
            PSTypeConstructorImpl(TypeConstructor)
              PSQualifiedImpl(Qualified)
                PSProperNameImpl(ProperName)
                  PsiElement(proper name)('A')
                  PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSTypeImpl(Type)
            PSTypeAtomImpl(TypeAtom)
              PSTypeConstructorImpl(TypeConstructor)
                PSQualifiedImpl(Qualified)
                  PSProperNameImpl(ProperName)
                    PsiElement(proper name)('S')
                    PsiElement(whitespace)('\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PsiElement(identifier)('showN')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PsiElement(identifier)('f')
                PsiElement(whitespace)(' ')
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PsiElement(identifier)('a')
                PsiElement(whitespace)('\n\n')
      PSValueDeclarationImpl(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSNamedBinderImpl(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSValueImpl(Value)
          PSPrefixValueImpl(PrefixValue)
            PSVarImpl(Var)
              PSQualifiedImpl(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSStringLiteralImpl(StringLiteral)
              PsiElement(string)('"Done"')