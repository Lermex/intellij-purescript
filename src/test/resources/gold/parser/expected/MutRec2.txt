Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Prelude')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)(' ')
        PsiElement(()('(')
        PSASTWrapperElement(PositionedDeclarationRef)
          PSASTWrapperElement(ValueRef)
            PsiElement(identifier)('log')
        PsiElement())(')')
        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeConstructor)
            PSASTWrapperElement(Qualified)
              PsiElement(proper name)('B')
              PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('B')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('B')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeConstructor)
            PSASTWrapperElement(Qualified)
              PsiElement(proper name)('A')
              PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PsiElement(foreign)('foreign')
      PsiElement(whitespace)(' ')
      PsiElement(import)('import')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ExternDataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('S')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(FunKind)
          PsiElement(proper name)('Type')
          PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('f')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeConstructor)
              PSASTWrapperElement(Qualified)
                PsiElement(proper name)('A')
                PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('S')
                  PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('f')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Case)
              PsiElement(case)('case')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('a')
                      PsiElement(whitespace)(' ')
              PsiElement(of)('of')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(CaseAlternative)
                PSASTWrapperElement(Value)
                  PSASTWrapperElement(PrefixValue)
                    PSASTWrapperElement(Constructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('A')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('b')
                        PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Value)
                  PSASTWrapperElement(PrefixValue)
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('g')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('b')
                        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('g')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('b')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Case)
              PsiElement(case)('case')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Value)
                PSASTWrapperElement(PrefixValue)
                  PSASTWrapperElement(Var)
                    PSASTWrapperElement(Qualified)
                      PsiElement(identifier)('b')
                      PsiElement(whitespace)(' ')
              PsiElement(of)('of')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(CaseAlternative)
                PSASTWrapperElement(Value)
                  PSASTWrapperElement(PrefixValue)
                    PSASTWrapperElement(Constructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('B')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Value)
                  PSASTWrapperElement(PrefixValue)
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('f')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Var)
                      PSASTWrapperElement(Qualified)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('showN')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeConstructor)
              PSASTWrapperElement(Qualified)
                PsiElement(proper name)('A')
                PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('S')
                  PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('showN')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('f')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('a')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('log')
                PsiElement(whitespace)(' ')
            PSASTWrapperElement(StringLiteral)
              PsiElement(string)('"Done"')